operations/CSM_product_management/Run_NCN_Personalization.md

```diff
@@ -0,0 +1,167 @@
+
+1. Upload the configuration file to CFS to update or create the
+   `ncn-personalization` CFS configuration.
+   ```bash
+   ncn# cray cfs configurations update ncn-personalization --file ncn-personalization.json --format json
+   {
+      "lastUpdated": "2021-07-28T03:26:01Z",
+      "layers": [
+         { ... layer information here ... },
+      ],
+      "name": "ncn-personalization"
+   }
+   ```
+   > **WARNING**:
+   > When running the above step, the CFS session may fail due to the node's
+   > sshd configuration being corrupted by a different process. Run the
+   > following command on all NCNs to fix the issue.
+   >
+   > ```bash
+   > ncn# systemctl restart cfs-state-reporter
+   > ```
+
+<a name="ncn_personalization_set_component_config"></a>
+### Set the Desired Configuration on all NCNs
+
+1. Update the desired configuration for all NCNs.
+
+   ```bash
+   ncn# for xname in $(cray hsm state components list --role Management--format json | jq -r .Components[].ID)
+   do
+       cray cfs components update --desired-config ncn-personalization --enabled true --format json $xname
+   done
+   ```
+   After this command is issued, the CFS Batcher service will dispatch a CFS
+   session to configure the NCNs. Since the NCN is now managed by CFS by setting
+   a desired configuration, the same will happen every time the NCN boots.
+
+   > **WARNING**:
+   > When running the above step, the CFS session may fail due to the node's
+   > sshd configuration being corrupted by a different process. Run the
+   > following command on all NCNs to fix the issue.
+   >
+   > ```bash
+   > ncn# systemctl restart cfs-state-reporter
+   > ```
+
+1. Query the status of the NCN Personalization process. The status will be
+   `pending` while the node is being configured by CFS, and will change to
+   `configured` when the configuration has completed.
+
+   ```bash
+   ncn# export CRAY_FORMAT=json
+   ncn# for xname in $(cray hsm state components list --role Management | jq -r .Components[].ID)
+   do
+       cray cfs components describe $xname | jq -r ' .id+" status="+.configurationStatus'
+   done
+   x3000c0s17b0n0 status=configured
+   x3000c0s19b0n0 status=pending
+   x3000c0s21b0n0 status=configured
+   ...
+   ```
+
+   The NCN personalization step is complete and the NCNs are now configured as
+   specified in the `ncn-personalization` configuration layers.
+
+   See [Configuration Management of System Components](../configuration_management/Configuration_Management_of_System_Components.md)
+   for information on setting desired configuration on specific nodes with CFS.
+
+<a name="rerun_ncn_personalization"></a>
+## Re-running NCN Personalization
+
+If no changes have been made to the configuration layers (such as a new layer,
+different playbook, or new commit made), but NCN personalization needs to be
+run again, CFS can re-run NCN personalization.
+
+Rerun the configuration for an NCN by clearing the state of the node. Clearing
```

Comment:
Nit: some places use rerun and others re-run.  Pick one.

Reply:
fixed